### PR TITLE
feature/spaceship: Clause 28: Localization, Clause 29: Input/output

### PR DESCRIPTION
**Pull request by statementreply**
Works towards #62.

### CODE REVIEW

File: stl/inc/filesystem [hunk 1]
```diff
@@ -2453,6 +2469,11 @@ namespace filesystem {
             return _Path == _Rhs._Path;
         }
 
+#if _HAS_CXX20
+        _NODISCARD strong_ordering operator<=>(const directory_entry& _Rhs) const noexcept {
+            return _Path <=> _Rhs._Path;
+        }
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
```
Comment (CaseyCarter) [inline stl/inc/filesystem:2476]: Shouldn't we move `operator<` from line 2464 into the `#else` branch of this preprocessor conditional as well? (-1 to whoever put the operators in this non-standard order in the first place.)

**Review by CaseyCarter — APPROVED** [review]

**Comment by CaseyCarter** [comment]
FYI: the test failure is a libc++ test for `directory_entry`'s comparison operators that expects actual member functions (e.g., `operator<=`) to be present. I've pushed a fix upstream, and opened #1594 to update our LLVM submodule reference to get the fixed test. Once #1594 merges, we'll need to merge it to feature/spaceship which we can then merge into this PR.

**Review by CaseyCarter — CHANGES_REQUESTED** [review]
Oops - I meant to "request changes" for my `directory_entry::operator<` comment.

**Review by CaseyCarter — APPROVED** [review]

**Comment by CaseyCarter** [comment]
Moving this to final review despite the test failure which is addressed by #1594. If there is a problem with #1594 and we absolutely must merge this first, we can skip the failing libc++ test in this PR and re-enable in #1594.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for spaceship-izing locales and I/O so we can properly communicate with aliens! :alien: :speech_balloon: :satellite:

**Pull request merged**